include/oxen/quic/address.hpp

```diff
@@ -340,6 +342,12 @@ namespace oxen::quic
             return *this;
         }
 
+        auto operator<=>(const Path& other) const { return std::tie(local, remote) <=> std::tie(other.local, other.remote); }
```

Comment:
`==` (and `!=`) make sense to me, but I'm less convinced that general lexicographic ordering make sense on a Path object.  To elaborate, if I have two paths `p1` and `p2`, then `p1 < p2` doesn't have an obvious meaning.  It's a bit like `std::complex<double>` in that regard: yes, you *could* define inequality operations, but they wouldn't have an obvious meaning.

Reply:
It made sense to me when thinking about a local endpoint using paths as key that lead to different remotes:
```c++
Address local_addr{...};

Path path1{local_addr, {remote_addr1, 5555};
Path path2{local_addr, {remote_addr2, 5555}};
Path path3{local_addr, {remote_addr2, 7777}};
```

Using `std::tie` will compare `local` vs `other.local`, then `remote` vs `other.remote` using the `operator<=>` already defined for the `Address` object.

This would result in:

```c++
(path2 > path1) == true;
(path3 > path1) != true;
(path3 > path2) == true;
```

What do you think?

Reply:
A local and remote is essentially a 2-dimensional object: you can think of `local` as the `x` dimension and `remote` as the `y` dimension, but an inequality operator between two such instances is a one-dimensional concept that you're applying to two-dimensional space.  You *can* define an order in such a space (such as lexicographic ordering of x-then-y), but it's an arbitrary definition rather than a natural one.

In particular, while x-then-y (i.e. local-then-remote) is indeed *a* valid ordering, it isn't intuitive to me that it is an ordering that is usually preferable to all other orderings.  Lexicographic y-then-x ordering, for instance, seems like another worthy contender, but there are other possibilities (for instance: putting paths with all-private addresses before ones with public components).  When you have more than one different, but reasonable, orderings I don't think it makes sense to define order at all; code that wants to order instances of the objects is then responsible for figuring out the ordering that makes sense in its context.